src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/ActionBarViewModel.cs

```diff
@@ -31,6 +31,19 @@ public ListItemViewModel? SelectedItem
     [ObservableProperty]
     public partial bool ShouldShowContextMenu { get; set; } = false;
 
+    [ObservableProperty]
+    public partial string PageName { get; private set; } = string.Empty;
+
+    public PageViewModel? CurrentPage
+    {
+        get => field;
+        set
+        {
+            field = value;
+            this.PageName = field?.Name ?? string.Empty;
+        }
+    }
+
```

Comment:
I'd make `CurrentPage` the observable property here as it's set from the outside. then make `public PageName { get; private set; } = CurrentPage?.Name ?? string.Empty;` a standard property and use [`NotifyPropertyChangedFor(nameof(PageName))`](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/observableproperty#notifying-dependent-properties) on the CurrentPage so when it updates it recalculates the PageName for updating the UI binding. This avoids the special logic in the setter.

This also gives you the partial method `OnCurrentPageChanging/Changed` in order to respond to that to do other things/setup in the ViewModel here like if the selected item needs updating based on the new page, etc...

Reply:
Okay I agree this is cleaner. However, after I made this change, it actually raised _more_ questions. Now, when we navigate deeper, we immediately raise the `PropertyChanged(nameof(PageName))` - that makes sense. But at that moment, the `PageViewModel` hasn't necessarily loaded the `Name` out of the extension yet. So later on when we do fetch the name, the PropertyChanged(Name) doesn't bubble all the way up to the XAML. Any magic community toolkit helpers for that scenario?

Reply:
Ah, then in that case we want to just probably be binding directly to `{x:Bind CurrentPage.Name, Mode=OneWay}` wherever this is used in the XAML. If both `CurrentPage` here and `Name` in the `PageViewModel` are both observable properties, then it should properly update the UI once that propertychanged event fires.